projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,234 @@
+            @Override
+            public void onClick(View view) {
+                createPdf();
+            }
+        });
+
+    }
+
+    private void createPdf() {
+
+        // Activate iText license
+        // TODO: Find a better solution for reading the licensekey file (This was a just a "quick fix" to get it working).
+        InputStream licenseIs = getResources().openRawResource(R.raw.itextkey);
+        LicenseKey.loadLicenseFile(licenseIs);
+        try {
+            HtmlConverter.convertToPdf(fillTemplate(), new FileOutputStream(Paths.get(lawsuitOutputPath, getPdfFilename()).toString()));
+        } catch (IOException e) {
+            Log.w(TAG, "Error creating PDF Lawsuit");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Fill lawsuit template with user & spam company's details.
+     *
+     * @return Template's content as Html with fields replaced.
+     */
+    private String fillTemplate() {
+
+        String lawsuit = readTemplate();
+
+        // General form fields
+        lawsuit = lawsuit.replace("&lt;claimCase&gt;", sentHaser ? claimCaseHaser : claimCaseSubscription);
+        lawsuit = lawsuit.replace("&lt;claimAmount&gt;", claimAmount);
+        lawsuit = lawsuit.replace("&lt;spamType&gt;", spamType);
+        lawsuit = lawsuit.replace("&lt;moreThanFiveLawsuits&gt;", sentMoreThanFiveLawsuits ? moreThanFiveLawsuits : lessThanFiveLawsuits);
+        lawsuit = lawsuit.replace("&lt;undefined&gt;", "");
+        lawsuit = lawsuit.replace("&lt;receivedSpamDate&gt;", dateFormatter.format(receivedSpamDate));
+        lawsuit = lawsuit.replace("&lt;currentDate&gt;", dateFormatter.format(new Date()));
+
+        // User fields
+        lawsuit = lawsuit.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastName);
+        lawsuit = lawsuit.replace("&lt;userId&gt;", userID);
+        lawsuit = lawsuit.replace("&lt;userAddress&gt;", userAddress);
+        lawsuit = lawsuit.replace("&lt;userPhone&gt;", userPhone);
+        lawsuit = lawsuit.replace("&lt;userFax&gt;", userFax);
+
+        // First spam company
+        lawsuit = lawsuit.replace("&lt;companyName&gt;", companyName);
+        lawsuit = lawsuit.replace("&lt;companyId&gt;", companyId);
+        lawsuit = lawsuit.replace("&lt;companyAddress&gt;", companyAddress);
+        lawsuit = lawsuit.replace("&lt;companyPhone&gt;", companyPhone);
+        lawsuit = lawsuit.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second spam company
+        lawsuit = lawsuit.replace("&lt;company2Name&gt;", company2Name);
+        lawsuit = lawsuit.replace("&lt;company2Id&gt;", company2Id);
+        lawsuit = lawsuit.replace("&lt;company2Address&gt;", company2Address);
+        lawsuit = lawsuit.replace("&lt;company2Phone&gt;", company2Phone);
+        lawsuit = lawsuit.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return lawsuit;
+    }
+
+    private String readTemplate() {
+
+        StringBuilder contentBuilder = new StringBuilder();
```

Comment:
You can simplify using this:
https://stackoverflow.com/a/14169729

Reply:
Done.